runtime/include/sandbox_request.h

```diff
@@ -65,9 +69,10 @@ sandbox_request_log_allocation(struct sandbox_request *sandbox_request)
  * @return the new sandbox request
  */
 static inline struct sandbox_request *
-sandbox_request_allocate(struct module *module, char *arguments, int socket_descriptor,
-                         const struct sockaddr *socket_address, uint64_t request_arrival_timestamp,
-                         uint64_t admissions_estimate)
+sandbox_request_allocate(struct module *module, bool request_from_outside, ssize_t request_length, 
```

Comment:
OK, the parameters in this function are indeed too many.

Reply:
Yes, you can use either malloc/free memory or to avoid dynamic mem alloc, then just create a local struct instance like this:
```
struct sandbox_chained_input sci;
sci.request_length = request_length;
...
```

Reply:
To be clear, C "complex" types like structs are just pass by value like other types. It's not like a class in a high level language where it has to be on the heap. Here's a small REPL to show some of the implications of that! https://replit.com/@spmcbride1201/pass-options

Reply:
My concern for using malloc/free is the performance issue, I tend to use static variables that stay in the stack, that's @emil916 's example code. If this struct has some pointer (pointing to the dynamic memory) and is passed by value, I think the pointer is also copied and points to the same dynamic memory, that is fine, I think.